src/linearAlgebra/BlasLapack.h

```diff
@@ -93,11 +93,28 @@ namespace dftefe
       template <typename ValueType,
                 typename dftefe::utils::MemorySpace memorySpace>
       void
-      ascale(size_type               n,
-             ValueType               alpha,
-             const ValueType         x,
-             ValueType *             z,
-             BlasQueue<memorySpace> &BlasQueue);
+      ascale(size_type                     n,
+             ValueType                     alpha,
+             const ValueType               x,
+             ValueType *                   z,
+             LinAlgOpContext<memorySpace> &context);
+
+
+      /**
+       * @brief Template for performing \f$ z_i = x_i * y_i$
+       * @param[in] size size of the array
+       * @param[in] x array
+       * @param[in] y array
+       * @param[out] z array
+       */
+      template <typename ValueType,
+                typename dftefe::utils::MemorySpace memorySpace>
+      void
+      hadamardProduct(size_type                     n,
+                      const ValueType               x,
```

Comment:
I think there is a * missing infront of x (i.e., it should be const ValueType * x)

Reply:
Thanks for catching. I have fixed it now.